app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/ReceiverDeviceAsyncTask.java

```diff
@@ -0,0 +1,116 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.Toast;
+
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.R;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.ERROR;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.SENDING;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.SENT;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.FILE_TRANSFER_PORT;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.copyToOutputStream;
+
+/**
+ * Helper class for the local file sharing module.
+ *
+ * Once the handshake has successfully taken place, this async-task is used to receive files from
+ * the sender device on the FILE_TRANSFER_PORT port. No. of files to be received (and their names)
+ * are learnt beforehand during the handshake.
+ *
+ * A single Task is used for the entire file transfer (the server socket accepts connections as
+ * many times as the no. of files).
+ */
+class ReceiverDeviceAsyncTask extends AsyncTask<Void, Integer, Boolean> {
+
+  private static final String TAG = "ReceiverDeviceAsyncTask";
+
+  private WifiDirectManager wifiDirectManager;
+  private String incomingFileName;
+
+  public ReceiverDeviceAsyncTask(WifiDirectManager wifiDirectManager) {
+    this.wifiDirectManager = wifiDirectManager;
+  }
+
+  @Override
+  protected Boolean doInBackground(Void... voids) {
+    try (ServerSocket serverSocket = new ServerSocket(FILE_TRANSFER_PORT)) {
+      Log.d(TAG, "Server: Socket opened at " + FILE_TRANSFER_PORT);
+
+      final String KIWIX_ROOT = wifiDirectManager.getZimStorageRootPath();
```

Comment:
don't name a variable like a constant if it isn't a constant

Reply:
So all caps with underscores is only for global constants?

Reply:
ALL_CAPS_UNDERSCORE is for any constant, if it is static and final it is a constant, the visiblity modifier has no bearing on the properties: There exists one of it & its reference can not be changed.
This value is transient, it exists briefly.

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/4e7a2e03c2957184e4ac59c2d901d958fd752092#diff-5f231fa33ed53daa1f723546988806f6R49)